components/Scroller/Scroller.tsx

```diff
@@ -32,13 +13,94 @@ function Scroller(props: {
   children: ScrollerItem | ScrollerItem[];
   onePageAtATime?: boolean;
 }) {
+  const [hasScroll, setHasScroll] = useState(false)
+
+  type scrollVisibilityApiType = React.ContextType<typeof VisibilityContext>;
+
+  const apiRef = useRef({} as scrollVisibilityApiType)
+
+  const handleUpdate = (data: {
+    initComplete: boolean;
+    isFirstItemVisible: boolean;
+    isLastItemVisible: boolean;
+  }) => {
+    const { initComplete, isFirstItemVisible, isLastItemVisible } = data
+    setHasScroll(initComplete && (!isFirstItemVisible || !isLastItemVisible))
+  }
+
   return (
     <div className={styles.horizontalScrollContainer}>
-      <ScrollMenu itemClassName={props.onePageAtATime ? styles.scrollPage : ""}>
+      <ScrollMenu apiRef={apiRef} onUpdate={handleUpdate} LeftArrow={LeftArrow} RightArrow={RightArrow} itemClassName={props.onePageAtATime ? styles.scrollPage : ""}>
```

Comment:
**question**: do you use uppercase propnames for props that take components?

Reply:
In all the components I've authored nope. I usually just fall back on the propTypes to convey that information. The `react-horizontal-scrolling` component does however use uppercase for these components. Don't think I'm a fan tbh

Reply:
yea I'm not a fan either but if that library is using that convention lets stick with it